spring-cloud-starter-stream-sink-router/README.adoc

```diff
@@ -53,6 +51,24 @@ For more information, please see the "Routers and the Spring Expression Language
 Integration Reference manual
 http://docs.spring.io/spring-integration/reference/html/messaging-routing-chapter.html#router-namespace[Configuring (Generic) Router section].
 
+NOTE: Starting with SCSt (Spring Cloud Stream) 2.0 the message wire format for `json`, `text` and `xml` content types is `byte[]` not String.
+The SpEL expressions for those content-types will either need to convert the payload to `String` or use the `jsonPath`
+or `xmlPath` utils that handle byte array payloads.
```

Comment:
this is called `#xpath()` SpEL-function: https://docs.spring.io/spring-integration/reference/html/xml.html#xpath-spel-function.

Please, modify a `#jsonPath()` respectively and add a link for docs on the matter: https://docs.spring.io/spring-integration/reference/html/spel.html#spel-functions

Also, it would be great to mention how to deal with any other content type. E.g. register some bean for conversion which should be called from the routing expression or mention that it still can be done with the custom SpEL-function.

Thanks

Reply:
I will fix the xpath and jsonPath references but i feel reluctant to go into provide a detailed descriptions about injecting conversion beans. It feels wrong to dive into explanations about SCSt wire-format breaking changes in an issue that aims to fix a `cyclic` bean bug :)  

IMO such explanation deserves a separate issue as it affects more than one app starters (`splitter` comes to mind). I will open another issue to document this. Perhaps we can make it an embeddable block that can be included in all apps that have to deal with byte array payloads.

Reply:
Here is the new ticket: https://github.com/spring-cloud-stream-app-starters/app-starters-release/issues/166